File: backend/CDataset/levelOrder/levelOrder_11.cpp
class Solution {
public:
    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };

    void helper(TreeNode* root,vector<vector<int>>&ans,int level){
        if(root==NULL)return;
        if(level>=ans.size()){//this means this level has not been parsed before and is therefore absent in the ans variable
            ans.push_back(vector<int>{root->val});
        }
        else{//as level is already present, simply push to that level's vector
            ans[level].push_back(root->val);   
        }
        helper(root->left,ans,level+1);
        helper(root->right,ans,level+1);
        
    }
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>>ans;
        helper(root,ans,0);
        return ans;
    }
};

File: backend/dataset/levelOrder/levelorder_11.py
from dataset_server import TreeNode
from typing import List
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        q = [root]
        level = [[root.val]]
        while q:
            size = len(q)
            l = []
            while size:
                front = q.pop(0)
                size-=1

                if front.left:
                    q.append(front.left)
                    l+=[front.left.val]
                if front.right:
                    q.append(front.right)
                    l+=[front.right.val]
            if l:
                level.append(l)

        return level             
